options/options_test.go

```diff
@@ -4,47 +4,58 @@ import (
 	"testing"
 )
 
-type ListOptions struct {
-	Offset                *int    `param:"offset"`
+type TestPaginationOptions struct {
+	Limit  *int `param:"limit"`
+	Offset *int `param:"offset"`
+}
+
+type TestListOptions struct {
+	TestPaginationOptions
 	MacAddress            *string `param:"macAddress"`
 	PrivateNetworkCapable *bool   `param:"privateNetworkCapable"`
 }
 
 func TestEncode(t *testing.T) {
 	testCases := []struct {
 		name     string
-		opts     ListOptions
+		opts     TestListOptions
 		expected string
 	}{
 		{
-			name: "Test Case 1: Full options",
-			opts: func() ListOptions {
+			name: "Full options",
+			opts: func() TestListOptions {
 				macAddress := "AA:BB:CC:DD:EE:FF"
 				offset := 0
+				limit := 10
 				privateNetworkCapable := false
-				return ListOptions{
```

Comment:
You didn't need to remove the tests you had before, they were valid test cases to have.
With the addition of your new feature, you just needed to add some new tests.

Reply:
Done